### ISSUE TITLE
[Accessibility], WCAG 4.1.2. Modal dialogs are missing a proper role and accessible names

### ISSUE THREAD

**Issue by Eiskore** [NatLibFi/Skosmos#1989]
## Description of the problem
The Suggest a new concept to the vocabulary form and Suggest a change to the concet are modal dialogs. Dialogs haven't been implemented technically according to accessibility requirements. 

Fix: A role (role="dialog") and an accessible name must be added to the dialog. The name is added either with aria-labelledby or aria-label attributes. Usually, the heading of the dialog serves as a good name for the dialog.

Severity: moderate
Priority: medium

## Additional information (e.g. screenshots) about the problem

<img width="1004" height="505" alt="Image" src="https://github.com/user-attachments/assets/39005966-66fd-4347-9e33-744a2b46cb6f" />

## The browser you used when the problem appeared

Chrome, Firefox, Edge
